File: Jack/Assets/HalloweenRange/_Scripts/Js/MuffGenerator.js
//MuffGenerator
var muff : Rigidbody;	    			// Object to instanciate
var muffSpeed: float = 1.0;				// Initial speed
var RotationCorrector: float = -90;		// Correcting muff rotation (set 0 to keep rotation unchanged)

private var nextFire = 0.0;


//============================================================================================

function GenerateMuff () {

 // Create muff and push it forward
   var muffClone : Rigidbody = Instantiate(muff, transform.position, transform.rotation);
   muffClone.transform.Rotate(Vector3.up, RotationCorrector);
   muffClone.velocity = transform.forward * muffSpeed;
}

//----------------------------------------------------------------------------------------------

File: Jack/Assets/HalloweenRange/_Scripts/C#/MuffGenerator.cs
using UnityEngine;
using System.Collections;

public class MuffGenerator : MonoBehaviour 
{
	public Rigidbody 	muff;	    					// Object to instanciate
	public float 		muffSpeed 			= 1.0f;		// Initial speed
	public float 		RotationCorrector 	= -90f;		// Correcting muff rotation (set 0 to keep rotation unchanged)

	void GenerateMuff () 
	{
		// Create muff and push it forward
		Rigidbody muffClone = Instantiate(muff, transform.position, transform.rotation)as Rigidbody;
		muffClone.transform.Rotate(Vector3.up, RotationCorrector);
		muffClone.velocity = transform.forward * muffSpeed;
		if(!muffClone.gameObject.active) 
			muffClone.gameObject.SetActiveRecursively(true);
	}
}
